### PR TITLE
Message Fix Fix Mediation Policies (xml_validator) failing with Global Mediation Extensions with FULL log

### PR DESCRIPTION
**Pull request by Sarangan0219**
Fixes https://github.com/wso2/carbon-apimgt/issues/8411

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java [hunk 1]
```diff
@@ -422,24 +423,38 @@ public static boolean handleThreat(org.apache.synapse.MessageContext messageCont
...
+        String contentType = axis2MC.getProperty(ThreatProtectorConstants.CONTENT_TYPE).toString();
+
         if (bufferedInputStream != null) {
-            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-            byte[] buffer = new byte[requestBufferSize];
-            int length;
-            while ((length = bufferedInputStream.read(buffer)) > -1) {
-                byteArrayOutputStream.write(buffer, 0, length);
+            bufferedInputStream.mark(0);
+            if (bufferedInputStream.read() != -1){
```
Comment (npamudika) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java:431]: Shall we fix the formatting issue here?

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java [hunk 2]
```diff
@@ -422,24 +423,38 @@ public static boolean handleThreat(org.apache.synapse.MessageContext messageCont
...
+                    byteArrayOutputStream.write(buffer, 0, length);
+                }
+                byteArrayOutputStream.flush();
+                inputStreamSchema = new ByteArrayInputStream(byteArrayOutputStream.toByteArray());
+                inputStreamXml = new ByteArrayInputStream(byteArrayOutputStream.toByteArray());
+                inputStreamOriginal = new ByteArrayInputStream(byteArrayOutputStream.toByteArray());
+                inputStreamJSON = new ByteArrayInputStream(byteArrayOutputStream.toByteArray());
+            } else {
+                String payload;
+                if (ThreatProtectorConstants.APPLICATION_JSON.equals(contentType)){
```
Comment (npamudika) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java:446]: Shall we fix the formatting issue here?

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java (hunk 1 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java:431]: Fixed in the latest Version

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java (hunk 2 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/utils/GatewayUtils.java:446]: Fixed in the latest Versions

**Pull request merged**